### PR TITLE
OCPBUGS-18072: Set emptyDir storage for the image registry only on initial time for None and Kubevirt platform

### PR DESCRIPTION
**Pull request by dharaneeshvrd**
This allows the user to override with custom storage for image registry

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/registry/registry.go [hunk 1]
```diff
@@ -25,9 +25,9 @@ func ReconcileRegistryConfig(cfg *imageregistryv1.Config, platform hyperv1.Platf
 	if cfg.Spec.HTTPSecret == "" {
 		cfg.Spec.HTTPSecret = generateImageRegistrySecret()
 	}
-	if (platform == hyperv1.KubevirtPlatform || platform == hyperv1.NonePlatform) &&
-		cfg.Spec.Storage.EmptyDir == nil {
 
+	storageNotSet := cfg.Spec.Storage == imageregistryv1.ImageRegistryConfigStorage{}
+	if (platform == hyperv1.KubevirtPlatform || platform == hyperv1.NonePlatform) && storageNotSet {
```
Comment (davidvossel) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/registry/registry.go:30]: should "NonePlatform" be "AgentPlatform" instead?

Comment (csrwng) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/registry/registry.go:30]: I'd stick to None for now unless we explicitly need to change the Agent one

**Review by bryan-cox — APPROVED** [review]
/lgtm

**Pull request merged**